experimentation/src/main/java/com/automattic/android/experimentation/local/FileBasedCache.kt

```diff
@@ -0,0 +1,50 @@
+package com.automattic.android.experimentation.local
+
+import com.automattic.android.experimentation.domain.Assignments
+import com.automattic.android.experimentation.remote.AssignmentsDtoJsonAdapter
+import com.automattic.android.experimentation.remote.AssignmentsDtoMapper.toAssignments
+import com.automattic.android.experimentation.remote.AssignmentsDtoMapper.toDto
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+
+internal class FileBasedCache(
+    private val cacheDir: File,
+    private val moshi: Moshi = Moshi.Builder().build(),
+    private val jsonAdapter: AssignmentsDtoJsonAdapter = AssignmentsDtoJsonAdapter(moshi),
+) {
+
+    private val assignmentsFile = File(cacheDir, "assignments.json")
+    private val type = Types.newParameterizedType(
+        Map::class.java,
+        Long::class.javaObjectType,
+        String::class.java,
+    )
+    private val wrapperAdapter = moshi.adapter<Map<Long, String>>(type)
+
+    suspend fun getAssignments(): Assignments? {
+        return withContext(Dispatchers.IO) {
+            assignmentsFile.takeIf { it.exists() }?.readText()?.let { json ->
+                val fromJson = wrapperAdapter.fromJson(json) ?: return@let null
+
+                val (fetchedAt, dtoJson) = fromJson.toList().first()
+
+                val dto = jsonAdapter.fromJson(dtoJson)
+
+                dto?.toAssignments(fetchedAt)
+            }
+        }
+    }
+
+    suspend fun saveAssignments(assignments: Assignments) {
+        withContext(Dispatchers.IO) {
+            val (dto, fetchedAt) = assignments.toDto()
+            val dtoJson = jsonAdapter.serializeNulls().toJson(dto)
+
+            val wrapperJson = wrapperAdapter.toJson(mapOf(fetchedAt to dtoJson))
+            assignmentsFile.writeText(wrapperJson)
+        }
+    }
```

Comment:
Question (❓): On the [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android) side, there is this [clearCachedAssignments()](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/3f240ecc9161efca1d75183a29640bd21db95cb6/fluxc/src/main/java/org/wordpress/android/fluxc/store/ExperimentStore.kt#L54-L56) implementation as well, is that not needed for our case here?

Reply:
Good catch, thanks! I added this with my next work, which I have locally. I think it's not worth to cherry-pick it here, it'll land with other changes

Reply:
Awesome, thanks! 👍